internal/errors/net_test.go

```diff
@@ -0,0 +1,239 @@
+package errors
+
+import (
+	"math"
+	"testing"
+	"time"
+)
+
+func TestErrFailedInitDialer(t *testing.T) {
+	type want struct {
+		want error
+	}
+	type test struct {
+		name       string
+		want       want
+		checkFunc  func(want, error) error
+		beforeFunc func()
+		afterFunc  func()
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return an ErrFailedInitDialer error",
+			want: want{
+				want: New("failed to init dialer"),
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(tt *testing.T) {
+			if test.beforeFunc != nil {
+				test.beforeFunc()
+			}
+			if test.afterFunc != nil {
+				defer test.afterFunc()
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			got := ErrFailedInitDialer
+			if err := test.checkFunc(test.want, got); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+		})
+	}
+}
+
+func TestErrInvalidDNSConfig(t *testing.T) {
+	type args struct {
+		dnsRefreshDur time.Duration
+		dnsCacheExp   time.Duration
+	}
+	type want struct {
+		want error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return an ErrInvalidDNSConfig when dnsRefreshDur is 5 minute, dnsCacheExp is 4 minute",
+			args: args{
+				dnsRefreshDur: 5 * time.Minute,
+				dnsCacheExp:   4 * time.Minute,
+			},
+			want: want{
+				want: New("dnsRefreshDuration  > dnsCacheExp, 5m0s, 4m0s"),
+			},
+		},
+		{
+			name: "return an ErrInvalidDNSConfig when dnsRefreshDur is 0, dnsCacheExp is 0",
+			args: args{
+				dnsRefreshDur: 0,
+				dnsCacheExp:   0,
+			},
+			want: want{
+				want: New("dnsRefreshDuration  > dnsCacheExp, 0s, 0s"),
+			},
+		},
+		{
+			name: "return an ErrInvalidDNSConfig when dnsRefreshDur and dnsCacheExp are the minimum number of int64",
+			args: args{
+				dnsRefreshDur: time.Duration(math.MinInt64),
+				dnsCacheExp:   time.Duration(math.MinInt64),
+			},
+			want: want{
+				want: Errorf("dnsRefreshDuration  > dnsCacheExp, %s, %s", time.Duration(math.MinInt64), time.Duration(math.MinInt64)),
+			},
+		},
+		{
+			name: "return an ErrInvalidDNSConfig when dnsRefreshDur and dnsCacheExp are the maximum number of int64",
+			args: args{
+				dnsRefreshDur: time.Duration(math.MaxInt64),
+				dnsCacheExp:   time.Duration(math.MaxInt64),
+			},
+			want: want{
+				want: Errorf("dnsRefreshDuration  > dnsCacheExp, %s, %s", time.Duration(math.MaxInt64), time.Duration(math.MaxInt64)),
+			},
+		},
+	}
```

Comment:
- Could you add below case?
    1. when dnsRefreshDur is nil.
    2. when dnsCacheExp is nil.
    3. when dnsRefreshDur and dnsCacheExp is nil

Reply:
@vankichi thank you for your comment.
`time.Duration` is a primitive type, so I think we can not add a test case of nil pattern. https://play.golang.org/p/xdI5op3KTUu

From the above, instead of the nil case pattern, I added zero value pattern 🤔 
what do you think? 
if I made a mistake, I am sorry.

Reply:
thanks.
So I think we have to add below patterns
1. when dnsRefreshDur is 0.
2. when dnsCacheExp is 0.

and you already add the test when both of these value is 0, it'd be better renaming `return an ErrInvalidDNSConfig when dnsRefreshDur is default value(or some  better word), dnsCacheExp is default value (or some more better word) ` from `return an ErrInvalidDNSConfig when dnsRefreshDur is 0, dnsCacheExp is 0`.
also, it'd be better clearly if you remove these values definition from `args{}`.